src/legacy/core_plugins/interpreter/server/lib/route_expression/thread/index.js

```diff
@@ -37,12 +37,14 @@ export function getWorker() {
     if (type === 'run') {
       const { threadId } = msg;
       const { ast, context } = value;
-      heap[threadId]
-        .onFunctionNotFound(ast, context)
-        .then(value => {
-          worker.send({ type: 'msgSuccess', id, value: value });
-        })
-        .catch(e => heap[threadId].reject(e));
+      if (heap[threadId]) {
```

Comment:
how is it possible that the heap[threadId] does not exist ?

Reply:
I don't fully grok it at the moment, as mentioned in chat I'd like to learn more about the `heap` concept here, will reach out to @w33ble who's most familiar with the code. 

The error was caused by acting on an object that isn't in this lookup table, so it's safe to avoid that and with that, the server crash.

Reply:
This came up in one of Peter's PRs too, but he thought at the time it was related to some other change. I think @chrisdavies did an awesome job explaining what happens [in this comment](https://github.com/elastic/kibana/issues/29035#issuecomment-456421405)

> What's happening is that `worker.js` is getting lots of calls to `onFunctionNotFound`, each with the same `threadId`, but with different params

This happens because something like `mapColumn` will map over every row, as you can see [in the example Stacey provided in the original issue](https://github.com/elastic/kibana/issues/29035#issuecomment-455877678)

> Once the first one fails, the `threadId` is removed from the heap in `thread/index.js`, so the subsequent calls fail to find any heap context for that `threadId`.

And that's why it was blowing up. In Stacey's example, she was using `doesntexist` (an invalid function) as the function to run in `mapColumn`. That rejects, clears the heap, and subsequent heap actions blow up because the id is no longer valid.